### ISSUE TITLE
Campaign field in analytics cannot be set

### ISSUE THREAD

**Issue by XavierMelladoAtIncubapps** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#247]
Hi!

When I build a link, I set the **campaign** value in analytics to a given value. But, when I use the link, this field is always empty, no matter what I initially set. I think this issue is related to #162.

For us, not setting the campaign is a real show stopper. Please, can anyone help me? Am I doing something wrong or has the Cordova plugin a bug?

Thanks in advance,
Xavi

I am using Ionic 2 Beta 11. This is my system information:

```
Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.0.0-beta.37
Ionic App Lib Version: 2.0.0-beta.20
ios-deploy version: 1.9.0 
ios-sim version: 5.0.11 
OS: Mac OS X El Capitan
Node Version: v4.6.2
Xcode version: Xcode 8.1 Build version 8B62 
```

I am using the latest version of the plugin:

`<plugin name="io.branch.sdk" spec="~2.4.2">`

**Comment by renesansz** [comment]
Hey @XavierMelladoAtIncubapps ,
Thanks for reporting, It seems the `campaign` field was not implemented in the Cordova plugin yet.
I'll submit a PR soon and hopefully will resolve this issue.

Cheers,

**Comment by renesansz** [comment]
submitted PR #249 to address this issue

**Comment by ethanneff** [comment]
fixed in #255.
